srtcore/buffer.cpp

```diff
@@ -523,7 +523,19 @@ int CSndBuffer::readData(const int offset, srt::CPacket& w_packet, steady_clock:
     // XXX Suboptimal procedure to keep the blocks identifiable
     // by sequence number. Consider using some circular buffer.
     for (int i = 0; i < offset; ++i)
+    {
+        if (p == m_pLastBlock)
+        {
+            LOGC(qslog.Error, log << "CSndBuffer::readData: offset " << offset << " too large!");
+            return 0;
+        }
```

Comment:
```suggestion
    for (int i = 0; i < offset && p != m_pLastBlock; ++i)
    {
```

Reply:
@gou4shi1 Hope you don't mind me going ahead and merging the suggestion to proceed.

Reply:
of coz, thx for your suggestion